libsawtooth/src/client/rest.rs

```diff
@@ -149,6 +153,118 @@ impl SawtoothClient for RestApiSawtoothClient {
+            SawtoothClientError::new_with_source("unable to parse file contents", err.into())
+        })?;
+        let len = batch_list.batches.len();
+        let batch_ids = batch_list
+            .batches
+            .iter()
+            .map(|batch| batch.header_signature.clone())
+            .collect();
+
+        let batch_lists = split_batches(batch_list, size_limit);
+
+        let start = Instant::now();
+
+        for list in batch_lists.into_iter() {
+            let submit_list = list.write_to_bytes().map_err(|err| {
+                SawtoothClientError::new_with_source(
+                    "failed to get bytes from batch list",
+                    err.into(),
+                )
+            })?;
+            let request = Client::new()
+                .post(&url)
+                .header(header::CONTENT_TYPE, "application/octet-stream")
+                .body(submit_list);
+            request.send().map_err(|err| {
+                SawtoothClientError::new_with_source("request failed", err.into())
+            })?;
+        }
+        println!(
+            "batches: {},  batch/sec: {}",
+            len,
+            len as f64 / (((start.elapsed()).as_millis() as f64) / 1000f64)
+        );
+
+        if let Some(wait) = wait {
+            wait_for_batches(&self.url, batch_ids, wait)
+        } else {
+            Ok(())
+        }
+    }
+}
+
+fn wait_for_batches(
+    url: &str,
+    ids: Vec<String>,
+    wait: Duration,
+) -> Result<(), SawtoothClientError> {
+    let end_time = Instant::now()
+        .checked_add(wait)
+        .ok_or_else(|| SawtoothClientError::new("failed to schedule timeout"))?;
+    let start_time = Instant::now();
+    loop {
+        let url = format!(
+            "{}/batch_statuses?id={}&wait={:?}",
+            url,
+            ids.join(","),
+            wait.as_secs()
+        );
+        let error_msg = "unable to get statuses";
+        let statuses = get::<StatusList>(&url, error_msg)?;
+
+        if let Some(status_list) = statuses {
+            let any_pending_batches = status_list
+                .data
+                .iter()
+                .any(|info| &info.status == "PENDING");
+            if any_pending_batches {
+                if Instant::now() < end_time {
+                    continue;
+                } else {
+                    println!("wait timed out! Some batches have not yet been committed...\nid  {}\ninvalid_transactions  {:?}\nstatus  {}",
+                        status_list.data[0].id, status_list.data[0].invalid_transactions, status_list.data[0].status);
+                    return Ok(());
+                }
+            } else {
+                println!("all batches committed in {:?}", start_time.elapsed());
+                return Ok(());
+            }
+        } else {
+            return Err(SawtoothClientError::new(
+                "could not retrieve batch statuses",
```

Comment:
I think this error message can be a bit more detailed; it should say that the batch statuses were not found (since a 404 is the cause).

Reply:
Fixed